tools/Makefile

```diff
@@ -164,9 +164,10 @@ tools-core += zstd
 $(foreach tool, $(tools-y), $(eval $(curdir)/$(tool)/compile += $(patsubst %,$(curdir)/%/compile,$(tools-core))))
 tools-y += $(tools-core)
 
-# make core tools depend on sed and flock
-$(foreach tool, $(filter-out zstd,$(tools-core)), $(eval $(curdir)/$(tool)/compile += $(curdir)/sed/compile))
-$(curdir)/zstd/compile += $(curdir)/flock/compile
+# make most core tools depend on sed and flock
+$(curdir)/patch/compile += $(curdir)/sed/compile
+$(curdir)/tar/compile += $(curdir)/sed/compile
+$(curdir)/zstd/compile += $(curdir)/libdeflate/compile $(curdir)/flock/compile
```

Comment:
I'm curious. Is flock needed?

Reply:
Well, now this interested me so I tried building it in a simple Alpine docker with just make, gcc and g++ and it builds without flock.

Reply:
well, busybox includes flock.

Reply:
Ah this is probably for macOS.

Reply:
I doubt ZSTD needs flock, removing the busybox applet symlink still doesnt break building

Reply:
Introduced with https://github.com/openwrt/openwrt/pull/14971/commits/23ba9fd813849dc2c8717374710bfca77ead2179 . xz might need flock. zstd does not.